crates/sqlexec/src/planner/session_planner.rs

```diff
@@ -947,6 +953,61 @@ impl<'a> SessionPlanner<'a> {
                 Ok(VariablePlan::ShowVariable(ShowVariable { variable }).into())
             }
 
+            // "DELETE FROM <table> WHERE <expression>"
+            //
+            // deletes rows from a table that matches the expression.
+            // or all the rows if no expression is provided.
+            ast::Statement::Delete {
+                tables,
+                from,
+                using: None,
+                selection,
+                returning: None,
+            } if tables.is_empty() => {
+                let (table_name, schema) = match from.len() {
+                    0 => {
+                        return Err(PlanError::InvalidDeleteStatement {
+                            msg: "DELETE FROM should have atleast one table name",
+                        })
+                    }
+                    1 => {
+                        let table_factor = from[0].relation.clone();
+                        let table_name = match table_factor {
+                            ast::TableFactor::Table { name, .. } => name,
+                            _ => {
+                                return Err(PlanError::UnsupportedFeature(
+                                    "DELETE from TableWithJoins",
+                                ))
+                            }
+                        };
+                        validate_object_name(&table_name)?;
+                        let table_name = object_name_to_table_ref(table_name)?;
+
+                        let table_source = context_provider
+                            .get_table_provider(table_name.clone())
+                            .await?;
+                        let table_source: &DefaultTableSource =
+                            table_source.as_any().downcast_ref().unwrap();
+                        let schema = table_source.table_provider.schema().to_dfschema()?;
```

Comment:
Since we just need the schema, we can avoid the downcast here:

```suggestion
                        let table_source = context_provider
                            .get_table_provider(table_name.clone())
                            .await?;
                        let schema = table_source.schema().to_dfschema()?;
```